ui-trackoccupancydiagram/src/components/helpers/drawElements/drawOccupancyZonesTexts.ts

```diff
@@ -35,71 +41,111 @@ export const drawOccupancyZonesTexts = ({
   };
   arrivalTime: number;
   departureTime: number;
+  yPosition: number;
+  isThroughTrain: boolean;
+  selectedTrainId: string;
 }) => {
   const zoneOccupancyLength = departureTime - arrivalTime - STROKE_WIDTH;
 
   const isBelowBreakpoint = (breakpoint: keyof typeof BREAKPOINTS) =>
     zoneOccupancyLength < BREAKPOINTS[breakpoint];
 
-  const textLength = ctx.measureText(zone.originStation!).width;
-  const { xName, yName } = {
-    xName: isBelowBreakpoint('medium')
-      ? arrivalTime - textLength + STROKE_WIDTH
+  const originTextLength = ctx.measureText(zone.originStation!).width;
```

Comment:
This `!` is not very safe, why do you need it ?
```suggestion
  const originTextLength = ctx.measureText(zone.originStation || '').width;
```

Reply:
done